lms/static/scripts/frontend_apps/errors.js

```diff
@@ -52,3 +65,44 @@ export class APIError extends Error {
     this.details = data.details;
   }
 }
+
+/**
+ * Does the current error object represent an authorization error or need
+ * to re-authorize from the back end?
+ *
+ * This is a special case. We're handling an APIError resulting from an API
+ * request, but there are no further details in the response body to guide us.
+ * This implicitly means that we're facing an authorization-related state.
+ *
+ * Put another way, if an APIError has neither an errorCode nor a serverMessage,
+ * it is considered an "authorization error".
+ *
+ * @param {Error} error
+ * @returns {boolean}
+ */
+export function isAuthorizationError(error) {
+  return error instanceof APIError && !error.serverMessage && !error.errorCode;
+}
+
+/**
+ * Does the current error object represent an API Error with a recognized
+ * backend-provided `errorCode` related to a failed attempt to launch an
+ * assignment?
+ *
+ * @param {Error} error
+ * @returns {error is APIError}
+ */
+export function isLTILaunchServerError(error) {
+  return (
+    error instanceof APIError &&
+    !!error.errorCode &&
```

Comment:
This check (`!!error.errorCode`) is technically redundant since the third clause will only evaluate to true if this clause is also true.

Reply:
Typescript error without it, though. (`Type 'string | boolean | undefined' is not assignable to type 'boolean'.
  Type 'undefined' is not assignable to type 'boolean'`)